### DISCUSSION TITLE
Does conduit support multi-tenant and RBAC?

### DISCUSSION THREAD

**Discussion by link89** [ConduitPlatform/Conduit#578] [Q&A] [answered]
Hi, I am currently evaluating BAAS frameworks for  internal tools (I am tired of repeating setup db, auth, etc for every projects).  Conduit came into my consideration because it is a relatively lightweight option compared to Supabase and is implemented with TypeScript. However, I did not find several features that I consider important in the documentation. 

1. I am wondering if Conduit supports multiple tenancies, meaning whether I can create multiple logically independent tenants within one Conduit deployment for different projects. 
2. I am interested in RBAC permission support. I hope to create different user roles within a project to restrict their data access (of course, it might be feasible to use alternative permission control solutions as well).

**Comment by kkopanidis** [comment, marked as answer]
Greetings @link89, thanks for your interest in Conduit!

1. Conduit doesn't support multi-tenancy and we have no plans to support it. It would introduce significant runtime overhead, and it really isn't in our list of goals. We are planning to offer a SaaS model though, that could provide the same experience/benefits but not on a self-hosted environment. 
2. We have RBAC support through the authorization module, and the authentication module is already connected with it in the latest versions. We're doing a major re-work as part of 0.16 so you should expect to see more modules working with it soon.

**Reply by link89** [reply]
> Conduit doesn't support multi-tenancy and we have no plans to support it. 

About multi-tenancy, if it is not supported, is there a way to create multiple projects within a deployment? 

> We have RBAC support through the authorization module

Is there some sample code about this? For example, how to use it to limit data access in REST API?

**Reply by kkopanidis** [reply]
Hello @link89, no there isn't a way to create multiple projects within a deployment. Adding "projects" as a concept wouldn't create any meaningful separation for you, since again all configuration, schemas, users etc would be shared.

For the authorization module there isn't currently any documentation/code, as I've mentioned we're re-reworking things, so by the end of it we'll also produce relevant documentation and samples.
